### ISSUE TITLE
Versioning

### ISSUE THREAD

**Issue by fmunteanu** [JohnSully/KeyDB#29]
I'm trying to understand the logic behind current versioning you use (e.g. 0.9.4 to 0.9.4-r2, instead of 0.9.5). I'm an Linux rpm packager and this type of versioning creates issues on the long term.

Regards,

Floren

**Comment by JohnSully** [comment]
Hi Floren. 

The short answer is I backed myself into a corner by starting at 0.9. The r releases are for the case where a major bug is fixed but otherwise there were no changes to functionality. 

Once we hit 1.0 the third digit will cover this case. The second digit will be for new features with no compatibility issues. And the major digit will be for major new features that are now stable.

**Comment by JohnSully** [comment]
I should also mention 1.0 is waiting on Advanced Replication gaining stability. Still lots of issues to iron out.

**Comment by fmunteanu** [comment]
Thanks for taking the time to post the details, I'm going to open a new ticket with the issues I have compiling KeyDB with `MALLOC=memkind` enabled.

**Comment by JohnSully** [comment]
I just tested memkind and it works for me (famous last words).

Make sure you have all the dependencies:
sudo apt install build-essential nasm autotools-dev autoconf libjemalloc-dev tcl tcl-dev uuid-dev

The main one being autotools-dev and autoconf.  The errors with memkind get buried and its quite hard to see the actual problem, but its almost always missing autotools and autoconf.

**Comment by fmunteanu** [comment]
@JohnSully this is why is bad practice to use this type of versioning:
```
Start: rpmbuild -bs
error: line 5: Illegal char '-' in: Version:                0.9.4-r2
```

**Comment by JohnSully** [comment]
Will it suffice if I do 0.9.4.2 for now?

**Comment by JohnSully** [comment]
Also just an FYI I'm a few days away from releasing 0.9.5.  

Most people try us through docker not compiling so it encourages frequent releases on my part.

**Comment by fmunteanu** [comment]
Not a problem, looking forward to 0.9.5 😄

**Comment by JohnSully** [comment]
0.9.5 is released!
